tls/s2n_client_key_exchange.c

```diff
@@ -35,7 +35,7 @@
 static int calculate_keys(struct s2n_connection *conn, struct s2n_blob *shared_key)
 {
     /* Turn the pre-master secret into a master secret */
-    GUARD(s2n_tls_prf_master_secret(conn, shared_key));
+    GUARD(s2n_kex_tls_prf(conn->secure.cipher_suite->key_exchange_alg, conn, shared_key));
```

Comment:
It looks strange to pass in the kex from inside conn alongside conn. This is because of reuse with the hybrid case where the kex is ambiguous, correct?

Reply:
Correct, and it follows the pattern of passing the struct to method that expects to use it like the rest of the s2n code base. The hybrid case isn't really a problem as it is currently written, the prf method will always use the highest level KEX's prf which would do the right thing.